docs/source/Schemas.md

```diff
@@ -41,14 +41,14 @@ first:
 
 Tables are the main way of defining objects in FlatBuffers, and consist
 of a name (here `Monster`) and a list of fields. Each field has a name,
-a type, and optionally a default value (if omitted, it defaults to `0` /
-`NULL`).
+a type, and optionally a default value. If omitted, it defaults to `0`
+for scalar types, or `NULL` for other types. Some languages support
+setting a scalar's default to `null`. This makes the scalar optional.
```

Comment:
I don't understand what you mean. Can you expand?

Reply:
I find having `NULL` and `null` confusing. I was wondering if we could keep `NULL` for reference types, and use `NotSet` for optional scalars.

Reply:
I like `null` better than `NotSet` because its a known keyword.

Would it be clearer if I just made all the "NULL" lowercase? They do mean the same thing to the schema,
`null == NULL == not present in table`. I could warn them that the generated code may have a few different strategies for representing values and references: There's pointers, the local option type, or even an extra `IsPresent` method

Reply:
I would also say to not use `NULL`, and only `null`.

As the documentation already says, there's 3 modes for a field, default, optional, and required. default is default for scalars, and optional for references. `null` is the default value for optional, though the language may use an actual optional type instead?

Reply:
Done (s/NULL/null)